test/unit/cmap/command_monitoring_events.test.js

```diff
@@ -0,0 +1,78 @@
+'use strict';
+
+const mock = require('../../tools/mock');
+const { connect } = require('../../../src/cmap/connect');
+const { Connection } = require('../../../src/cmap/connection');
+const { Msg, Query, GetMore, KillCursor } = require('../../../src/cmap/commands');
+const { CommandStartedEvent } = require('../../../src/cmap/command_monitoring_events');
+const { expect } = require('chai');
+const { Long } = require('bson');
+
+describe('Command Monitoring Events - unit/cmap', function () {
+  let server;
+  after(() => mock.cleanup());
+  before(() => mock.createServer().then(s => (server = s)));
+
+  it('should never hold references to commands passed into CommandStartedEvent objects', function (done) {
+    const commands = [
+      new Query('admin.$cmd', { a: { b: 10 }, $query: { b: 10 } }, {}),
+      new Query('hello', { a: { b: 10 }, $query: { b: 10 } }, {}),
+      new Msg('admin.$cmd', { b: { c: 20 } }, {}),
+      new Msg('hello', { b: { c: 20 } }, {}),
+      new GetMore('admin.$cmd', Long.fromNumber(10)),
+      new GetMore('hello', Long.fromNumber(10)),
+      new KillCursor('admin.$cmd', [Long.fromNumber(100), Long.fromNumber(200)]),
+      new KillCursor('hello', [Long.fromNumber(100), Long.fromNumber(200)])
+    ];
+
+    server.setMessageHandler(request => {
+      const doc = request.document;
+      if (doc.ismaster || doc.hello) {
+        request.reply(mock.DEFAULT_ISMASTER_36);
+      }
+
+      // blackhole all other requests
+    });
+
+    connect({ connectionType: Connection, hostAddress: server.hostAddress() }, (err, conn) => {
+      expect(err).to.be.undefined;
+      expect(conn).to.not.be.undefined;
```

Comment:
I think we can drop stubbing this connection part, and the mock server stuff if we pass:
```
{ id: 'someId', address: 'someHost' }
```
That looks like it would be accepted by the CommandStartedEvent constructor.